lib/mongodb.js

```diff
@@ -400,30 +406,30 @@ MongoDB.prototype.fromDatabase = function(model, data) {
     }
   }
 
-  data = this.fromDatabaseToPropertyNames(model, data);
+  data = this.fromDatabaseToPropertyNames(modelName, data);
 
   return data;
 };
 
 /*!
  * Convert JSON to database-appropriate format
  *
- * @param {String} model The model name
+ * @param {String} modelName The model name
  * @param {Object} data The JSON data to convert
  */
-MongoDB.prototype.toDatabase = function(model, data) {
-  const modelInstance = this._models[model].model;
-  var props = this._models[model].properties;
+MongoDB.prototype.toDatabase = function(modelName, data) {
+  const modelCtor = this._models[modelName];
+  const props = modelCtor.properties;
 
   if (this.settings.enableGeoIndexing !== true) {
-    visitAllProperties(data, modelInstance, coerceDecimalProperty);
+    visitAllProperties(data, modelCtor, coerceProperties);
     // Override custom column names
-    data = this.fromPropertyToDatabaseNames(model, data);
+    data = this.fromPropertyToDatabaseNames(modelName, data);
     return data;
   }
 
-  for (var p in props) {
-    var prop = props[p];
+  for (const p in props) {
+    const prop = props[p];
```

Comment:
While you are changing from `var` to `const`, I think it would be best to open a new pull request to update `eslint-config-loopback` to the latest version and let `eslint` to fix all `var` usages. See https://github.com/strongloop/eslint-config-loopback/releases/tag/v13.0.0